test/functional/core/path_spec.lua

```diff
@@ -129,4 +129,48 @@ describe('file search (gf, <cfile>)', function()
     test_cfile([[\\.\UNC\LOCALHOST\c$\temp\test-file.txt]], [[.]], [[\\.\UNC\LOCALHOST\c$\temp\test-file.txt]])
     test_cfile([[\\127.0.0.1\c$\temp\test-file.txt]], [[127.0.0.1]], [[\\127.0.0.1\c$\temp\test-file.txt]])
   end)
+
+  describe('using `finddir`', function()
+    it('with "\\ " on path', function()
+      local folder1 = 'folder name'
+      mkdir(folder1)
+      local folder2 = 'dir'
+      local expected_path = join_path(folder1, folder2)
+      mkdir(expected_path)
+      eq(expected_path, eval('finddir("'..folder2..'",fnameescape("'..folder1..'"))'))
+      rmdir(folder1)
+    end)
+
+    it('with "\\ " on name', function()
+      local folder1 = 'dir'
+      mkdir(folder1)
+      local folder2 = 'folder name'
+      local expected_path = join_path(folder1, folder2)
+      mkdir(expected_path)
+      eq(expected_path, eval('finddir(fnameescape("'..folder2..'"),"'..folder1..'")'))
+      rmdir(folder1)
```

Comment:
all of these tests look pretty similar, is there a way to de-dupe the code?

it also can be noisy to have many separate it() tests for each variation of an input. when testing many inputs, all the variations can live in one test (see `test_cfile` cases above).